More Platforms

andybak opened the pull request:
Pico builds. Non-VR builds. OpenXR across multiple targets (Steam Frame etc)

andybak committed:
First pass at updating build.yml to include more target platforms

[CI BUILD]

claude committed:
Fix OpenXR build failures for non-OpenXR platforms (MacOS, iOS)

This commit fixes build failures that occur when building for platforms
that don't support OpenXR (MacOS, iOS) by implementing platform-specific
compilation guards.

Changes:

1. CI Configuration (.github/workflows/build.yml):
   - Add OPENXR_SUPPORTED define to Windows OpenXR builds
   - Add OPENXR_SUPPORTED define to Android OpenXR builds
   - Remove com.unity.xr.openxr package from MacOS builds
   - Add all new platforms (Linux, MacOS, Android OpenXR) to the build matrix

2. Code Changes:
   - Wrap ControllerDeviceOpenXR.cs in #if OPENXR_SUPPORTED guards
   - Guard OpenXR controller instantiation in PeltzerController.cs
   - Guard OpenXR controller instantiation in PaletteController.cs

This ensures that OpenXR-specific code is only compiled for platforms
where OpenXR is actually supported and available, preventing both
compile-time and runtime errors on incompatible platforms.

andybak committed:
[CI BUILD]

andybak committed:
Disable Quest stuff in OpenXR

andybak committed:
[CI BUILD]

andybak committed:
Merge branch 'claude/fix-openxr-ci-platforms-011CUprpWadqWHs1mE4q9Uva' into feature/more-platforms

andybak committed:
Merge branch 'main' into feature/more-platforms

andybak committed:
Remove duplicate platform entries

andybak committed:
desktop menu was active by default

andybak committed:
Unity changed this

andybak committed:
[CI BUILD]

andybak committed:
Caches weren't updated when there were change to packages_to_remove

[CI BUILD]

andybak committed:
Small tweaks to build (typo and action version bumps)

andybak committed:
[CI BUILD]

andybak committed:
temp fix. don't fail if support dir is missing

andybak committed:
[CI BUILD]

andybak committed:
Remove open brush only step

andybak committed:
Slightly saner attempt at non-vr builds

andybak committed:
[CI BUILD]

andybak committed:
Minimal dummy implementations

andybak committed:
[CI BUILD]

andybak committed:
Always include UI shaders

andybak committed:
Apply main scene overrides to desktop prefab

andybak committed:
Clean up spectator cam and set initial params

andybak committed:
Update UnityGLTFSettings.asset

andybak committed:
Some unity auto changes

andybak committed:
[CI BUILD]

andybak committed:
Spectator camera was overriding XR camera

andybak committed:
XR controller type detection improvement

andybak committed:
Mac builds should await signing step. Remove linux build folder after zipping

mikeage commented on .github/workflows/build.yml:
```diff
@@ -394,22 +415,32 @@ jobs:
             androidVersionCode: "${{ needs.configuration.outputs.androidVersionCode }}${{ matrix.versionSuffix }}"
             version: ${{ needs.configuration.outputs.version }}
             buildName: ${{ needs.configuration.outputs.basename }}
+            buildsPath: build/${{ matrix.vrsdk }}
```
why do we need this?

mikeage commented on .github/workflows/build.yml:
```diff
@@ -642,24 +832,41 @@ jobs:
+          jinjanate Support/steam/main_depot.win.vdf.j2 > build_windows_openxr/main_depot.vdf
+          jinjanate Support/steam/main_depot.mac.vdf.j2 > build_macos/main_depot.vdf
+          jinjanate Support/steam/main_depot.linux.vdf.j2 > build_linux/main_depot.vdf
+          jinjanate Support/steam/installscript_win.vdf.j2 > build_windows_openxr/installscript_win.vdf
```
this file isn't relevant for Blocks

claude committed:
Fix Pico controller recognition by removing binding mask filter

The previous approach tried to detect specific controller types and apply
controller-specific binding masks, but this broke Pico controllers due to
timing issues with device enumeration.

The correct solution is simpler: don't filter bindings at all. Since Pico
controllers use the same button layout as Quest controllers, and the input
action bindings use generic OpenXR paths like <XRController>{RightHand}/trigger,
Unity's Input System will automatically map the correct inputs for any
standard XR controller.

This fix:
- Removes the Oculus Touch Controller binding mask filter
- Sets bindingMask to null to allow automatic controller detection
- Works for Pico, Quest, Vive, Index, and other OpenXR controllers

andybak committed:
Merge remote-tracking branch 'origin/claude/fix-platforms-issue-011CUxcdRnkKTvswjo2hxz26' into feature/more-platforms

# Conflicts:
#	Assets/Scripts/model/controller/ControllerDeviceOpenXR.cs

claude committed:
Add automatic controller detection for Pico and other XR controllers

Previously, the OpenXR controller implementation was hardcoded to use
Oculus Touch Controller bindings, even though Pico controller bindings
were defined in UnityXRInput.inputactions. This prevented Pico controllers
and other supported controllers from working correctly.

Changes:
- Added GetControllerBindingGroup() method that detects connected controller
  type by checking device name
- Updated InitAsBrush() and InitAsWand() to use detected controller bindings
  instead of hardcoded Oculus Touch bindings
- Added support for automatic detection of:
  - Pico controllers (PXR)
  - Valve Index controllers
  - HTC Vive controllers
  - Windows Mixed Reality (WMR) controllers
  - Zapbox controllers
  - Oculus/Meta Quest controllers (fallback)
- Added debug logging to help troubleshoot controller detection

This enables Pico controller support without requiring any configuration
changes, while maintaining backward compatibility with existing controllers.

claude committed:
[CI BUILD]

andybak committed:
Merge remote-tracking branch 'origin/claude/integrate-pico-controller-input-011CUzXvRghSRYJ9Xu9FZFng' into feature/more-platforms

claude committed:
Add enhanced debugging for Pico controller detection

Improved controller detection with comprehensive logging to diagnose
Pico controller input issues:

- Check InputSystem devices in addition to XR InputDevices for more
  reliable controller type detection
- Added device layout checking (looks for "pxr" in layout name)
- Check manufacturer field in addition to device name
- Retry device initialization if not valid on first attempt
- Log all InputSystem devices with their types and layouts
- Log the selected binding group and active bindings per action
- Warn when actions have no active controls

This will help diagnose why Pico controllers may not be responding
to input by showing exactly which devices are detected and which
bindings are active.

claude committed:
Add comprehensive Pico controller setup documentation

Controllers are visible but inputs don't work because the Pico OpenXR
interaction profile is not configured in the Unity project.

This documentation explains:
- What's already configured (input bindings, detection code)
- What's missing (Pico Unity Integration SDK/interaction profile)
- Three options to enable Pico support:
  1. Install Pico Unity Integration SDK (recommended)
  2. Use generic OpenXR profile as fallback
  3. Create custom interaction profile
- How to verify Pico support is working
- Troubleshooting common issues

The code changes already support Pico controllers, but Unity's OpenXR
runtime needs the Pico interaction profile to recognize PXR_Controller
devices and map their inputs.

claude committed:
Fix Pico controller input by removing device-specific binding masks

PROBLEM:
Controllers were visible (tracking worked) but inputs didn't respond.
The code was setting device-specific binding masks like:
  actionSet.bindingMask = InputBinding.MaskByGroup("Pico Controller");

This filtered bindings to ONLY vendor-specific device paths like
<PXR_Controller>, which don't exist without vendor SDKs. The generic
<XRController> bindings that work on all OpenXR devices were being
filtered out.

SOLUTION:
Remove all binding masks and let OpenXR automatically resolve bindings.
This enables vanilla OpenXR cross-platform support where a single build
works on all compliant devices (Pico, Quest, Vive, Index, WMR, etc.)
without vendor-specific SDKs.

CHANGES:
- ControllerDeviceOpenXR.cs:
  * Removed binding mask application in InitAsBrush()/InitAsWand()
  * GetControllerBindingGroup() now only logs device info (debug)
  * Added comments explaining vanilla OpenXR approach

- Documentation:
  * Renamed PICO_SETUP.md → OPENXR_CONTROLLERS.md
  * Rewritten to explain generic OpenXR binding approach
  * Removed vendor SDK installation instructions
  * Added troubleshooting for cross-platform builds

This allows the same APK/build to work on all OpenXR devices without
recompilation or device-specific configuration.

andybak committed:
Merge remote-tracking branch 'origin/claude/integrate-pico-controller-input-011CUzXvRghSRYJ9Xu9FZFng' into feature/more-platforms

andybak committed:
Revert "Fix Pico controller input by removing device-specific binding masks"

This reverts commit fdfe3b7c4d918f8ca191357e55a887abed5444a8.

andybak committed:
Revert "Add comprehensive Pico controller setup documentation"

This reverts commit 5b616364cd1e732d8c13e3fee348f70fd152556a.

andybak committed:
Revert "Add enhanced debugging for Pico controller detection"

This reverts commit cb5d0caf89685428152d0f18daa922d0de4507f5.

andybak committed:
Revert "Add automatic controller detection for Pico and other XR controllers"

This reverts commit d5236350dfd4ae05773211bf5e754f299460cce9.

andybak committed:
Revert a merge

andybak committed:
Revert a merge part 2

andybak committed:
Support files for MacOs build

andybak committed:
WIP cleanup of build.yml

andybak committed:
precommit cosmetic fix

andybak committed:
Fix build errors on non-XR platforms

andybak committed:
Workaround for tar failing

andybak committed:
Try removing the whole step

mikeage commented on .github/workflows/build.yml:
```diff
@@ -642,24 +821,41 @@ jobs:
           VERSION: ${{ needs.configuration.outputs.version }}
           OPEN_BLOCKS_APP_ID: ${{ vars.STEAM_APP_ID }}
           OPEN_BLOCKS_WINDOWS_DEPOT_ID: ${{ vars.STEAM_WINDOWS_DEPOT_ID }}
+          OPEN_BLOCKS_MAC_DEPOT_ID: ${{ vars.STEAM_MAC_DEPOT_ID }}
```
these need to be created in Steam

andybak committed:
Merge branch 'main' into feature/more-platforms

andybak committed:
Correct built apk name

k-lara committed:
Continue with default user config when unable to read/write

mikeage committed:
Fix signing paths, steam files, and permission preservation

mikeage reviewed (approved):
Tested briefly on Mac; seems to work (and is signed, from a branch where I disabled the checks)

k-lara committed:
Enable Meta Quest Support

k-lara committed:
Disable Meta Quest support and symmetric projection

andybak committed:
Add the pico profile

andybak committed:
Select Pico profile

andybak committed:
Merge branch 'feature/pico-input' into feature/more-platforms

andybak committed:
dotnet format

andybak committed:
Wrap pico profile in #if !XR_DISABLED

andybak committed:
Merge branch 'main' into feature/more-platforms

# Conflicts:
#	.github/workflows/build.yml
#	ProjectSettings/GraphicsSettings.asset

andybak committed:
Update actions

andybak committed:
Automatic settings updates

andybak committed:
Incorrect directory name

andybak committed:
Align the artifact name/path used here with the build job output

andybak committed:
Fix glob issues on MacOS

"readlink -f is not supported by the default macOS readlink, so this will fail on macos-latest. Use a macOS-compatible way to resolve the app path (e.g., realpath if available, or python -c/perl), or avoid -f entirely by globbing and basename-ing without dereferencing."

> Both instances replaced. The glob StandaloneOSX/OpenBlocks*.app expands to the matched path and basename strips the directory component — readlink -f was only serving as a roundabout way to materialise the glob, which basename handles directly.

andybak committed:
Added using UnityEngine.InputSystem; and replaced the two InputSystem.InputSystem.* calls with unqualified InputSystem.*. The namespace import makes the static class directly accessible without the redundant   double-qualification.

andybak committed:
We need the double qualified reference here

andybak committed:
Merge branch 'main' into feature/more-platforms

andybak committed:
Add Steam Android depot to CI release

andybak committed:
Minor code cleanup

andybak committed:
Bump CI action versions to match Open Brush

andybak committed:
Pico publish

andybak committed:
Stale doc comment
